pkg/logger/logger.go

```diff
@@ -1,7 +1,19 @@
 package logger
 
 type Logger interface {
+	MessageLogger
+	FieldLogger
+}
+
+type FieldLogger interface {
+	WithFields(fields ...interface{}) MessageLogger
```

Comment:
the problem with this approach is that the `MessageLogger` is part of the return type, meaning that if a consumer wanted to bring their own logger, that logger would need to have this interface in the signature, which wont scale.